menpo/model/pca.py

```diff
@@ -1566,3 +1674,114 @@ def __str__(self):
+        if camera_settings is not None:
+            mlab.move(*camera_settings[0])
+            mlab.view(*camera_settings[1])
+            mlab.roll(camera_settings[2])
+        else:
+            fig.scene.z_plus_view()
+
+        if isinstance(list_weights, int):
+            list_weights = [list_weights]
+
+        if isinstance(list_components, int):
+            list_components = [list_components]
+        list_components = np.asarray(list_components)
+        are_components = list_components < self.n_components
+        components_to_be_rendered = list_components[are_components]
+        for not_exist_component in list_components[~are_components]:
+            print('Component {} does not exist'.format(not_exist_component))
+
+        mesh = self.mean()
+        s = mlab.triangular_mesh(mesh.points[:, 0], mesh.points[:, 1],
+                                 mesh.points[:, 2], mesh.trilist,
+                                 color=mesh_color)
+
+        mlab.savefig(str(export_path / 'Mean face.{}'.format('png')))
+        for component in components_to_be_rendered:
+            parameters = np.zeros(component+1)
+            std = self.eigenvalues[component] ** 0.5
+            for weight in list_weights:
+                parameters[component] = std*weight
+                s.mlab_source.points = self.instance(parameters).points
+                full_filename = '{}_{}_{}.{}'.format(filename,
+                                                     component,
+                                                     weight,
+                                                     'png')
+                mlab.savefig(str(export_path / full_filename))
+
+    def to_list(self):
+        r"""
+        Returns the main attributes of a model as a list
+        for serialization with avatars.serialize_morphable_model_to_binary
+        User should only expand the list.
+
+        Returns
+        ----------
+        list_elements: A list with the following  elements:
+                       mean_points: (n_features,) float32 ndarray
+                                     Flatten array with the points of the mean
+                                     mesh, casted as float32
+
+                       trilist:     (n_faces*3,) ndarray
+                                     Flatten array with the faces of the mean
+                                     mesh, casted as np.uint32
+
+                       components:   (n_features*n_components,) float32 ndarray
+                                     Flatten array with the components of the
+                                     PCAModel, casted as float32
+
+                       eigenvalues: (n_components,) float32 ndarray
+                                     Flatten array with the square rooot of
+                                     eigenvalues of the PCAModel,
+                                     casted as float32
+        """
+        return [self.mean().points.flatten().astype('float32'),
```

Comment:
I don't mind you having this function here but it does a lot of unnecessary copying:

```
mean = self.mean()
return [mean.points.ravel().astype('float32'),
            mean.trilist.ravel().astype('uint32'),
            self.components.astype('float32'),
            np.sqrt(self.eigenvalues).astype('float32')]
```

Would be great if these kinds of changes also went into a different PR?

Reply:
Ok, we can discuss about this in another PR.